pkg/server/plugin/upstreamca/awssecret/awslib_fake.go

```diff
@@ -0,0 +1,61 @@
+package awssecret
+
+import (
+	"io/ioutil"
+	"net/http"
+
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/service/secretsmanager/secretsmanageriface"
+	"github.com/stretchr/testify/suite"
+)
+
+type fakeSecretsManagerClient struct {
+	suite.Suite
+	secretsmanageriface.SecretsManagerAPI
+
+	mockStorage map[string]string
+}
+
+func newFakeSecretsManagerClient() (secretsmanageriface.SecretsManagerAPI, error) {
+	svc := new(fakeSecretsManagerClient)
+
+	return svc, nil
+
+}
+
+func (sm *fakeSecretsManagerClient) GetSecretValueRequest(input *secretsmanager.GetSecretValueInput) (*request.Request, *secretsmanager.GetSecretValueOutput) {
+
+	retreq := new(request.Request)
+	httpReq, _ := http.NewRequest("POST", "", nil)
+
+	retreq.HTTPRequest = httpReq
+
+	resp := secretsmanager.GetSecretValueOutput{}
+	resp.ARN = nil
+
+	cert, err := ioutil.ReadFile("_test_data/keys/cert.pem")
```

Comment:
seems a little cleaner to initialize `mockStorage` inside of `newFakeSecretsManagerClient`.